sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/PTransform.java

```diff
@@ -143,6 +143,9 @@
  */
 public abstract class PTransform<InputT extends PInput, OutputT extends POutput>
     implements Serializable /* See the note above */, HasDisplayData {
+
+  private static final long serialVersionUID = 3383862966597863311L;
```

Comment:
Unfortunately Flink uses Java serialization to serialize checkpoints (via SerializableCoder) in legacy Read transform (needed for KafkaIO), so without stabilizing this we are unable to upgrade even with the SerializablePipelineOptions moved back.

Reply:
Ack. Hopefully we can clean this up in the future.

Reply:
The problem is that Flink needs to serialize the UnboundedSource (after split) into state to support legacy Read. I'm not sure what is the correct path of fixing this - we could use Kryo, but UnboundedSource actually is `Serializable`. How does Dataflow serialize the source? I suppose the same applies for ReadViaSDF as well.